TDL15-CloudFormation-Jenkins/cf-deploy-jenkins.yaml

```diff
@@ -0,0 +1,245 @@
+          # Install the latest stable version of Jenkins
+          sudo yum install -y jenkins
+          # check whether Jenkins is running, type:
+          # systemctl status jenkins
+          # check jenkins login: 
+          sudo useradd -d /var/lib/jenkins jenkins # login dir
+          usermod -s /bin/bash jenkins # shell use dir
+          # set_jenkins_port  
+          sudo sed -i "s/JENKINS_PORT=\"8080\"/JENKINS_PORT=\"${JenkinsPort}\"/g" /etc/sysconfig/jenkins
+          # sshconfig: 
+          sudo echo -e "Host *\n\tStrictHostKeyChecking no\n" >> /var/lib/jenkins/.ssh/config
+          # ssh-keyscan -H hostname.com >> /var/lib/jenkins/.ssh/known_hosts
+          # Generate SSH key: 
+          if [ -z ${SnapshotId} ] ;then sudo ssh-keygen -t rsa -b 2048 -C jenkins@jenkins -N "" -f /var/lib/jenkins/.ssh/id_rsa; sudo cat /var/lib/jenkins/.ssh/id_rsa.pub > /var/lib/jenkins/.ssh/authorized_keys; fi
+          # set permissions
+          sudo chown -R jenkins:jenkins /home/jenkins /var/lib/jenkins /var/cache/jenkins /var/log/jenkins; sudo chmod 0700 /var/lib/jenkins/.ssh; sudo chmod 0600 /var/lib/jenkins/.ssh/*
+          # firewall: 
+          sudo firewall-cmd --permanent --add-service=ssh; sudo firewall-cmd --permanent --add-port='${JenkinsPort}'/tcp; sudo firewall-cmd --reload
+          # start the Jenkins service and enable it to start on system boot
+          sudo systemctl start jenkins
+          sudo systemctl enable jenkins
+          # install git
+          yum install -y git
+          # install AWS CLI
+          curl "https://awscli.amazonaws.com/awscli-exe-linux-x86_64.zip" -o "awscliv2.zip"
+          unzip awscliv2.zip
+          sudo ./aws/install
+          # install python
+          sudo yum -y groupinstall "Development Tools"
+          # sudo yum -y install openssl-devel bzip2-devel libffi-devel
+          sudo yum -y install bzip2-devel expat-devel gdbm-devel \
+            ncurses-devel openssl-devel readline-devel wget \
+            sqlite-devel tk-devel xz-devel zlib-devel libffi-devel
+          sudo wget https://www.python.org/ftp/python/${PythonVersion}/Python-${PythonVersion}.tgz
+          tar -xf Python-${PythonVersion}.tgz
+          cd Python-${PythonVersion}
+          ./configure --enable-optimizations # Setup installation by running the configure script, performs a number of checks to make sure all of the dependencies on your system are present
+          make -j 4 # Start the Python 3.8 build process 
+          sudo make altinstall #install the Python binaries
+          alias python="/usr/local/bin/python3.8"
+          # intall boto3
+          pip3.8 install boto3
+          # install terraform 
+          sudo wget https://releases.hashicorp.com/terraform/${TerraformVersion}/terraform_${TerraformVersion}_linux_amd64.zip
+          unzip terraform_${TerraformVersion}_linux_amd64.zip && sudo rm terraform_${TerraformVersion}_linux_amd64.zip -f && sudo mv terraform /usr/local/bin/
+
```

Comment:
and delete downloaded files here.

Reply:
This command line has rm zip file in the middle.